lib/internal/main/watch_mode.js

```diff
@@ -44,8 +43,10 @@ const argsWithoutWatchOptions = [];
 for (let i = 0; i < process.execArgv.length; i++) {
   const arg = process.execArgv[i];
   if (StringPrototypeStartsWith(arg, '--watch')) {
-    if (!StringPrototypeIncludes(arg, '=')) {
-      i++;
+    i++;
+    const nextArg = process.execArgv[i];
+    if (nextArg && StringPrototypeStartsWith(nextArg, '--')) {
```

Comment:
this can be problematic, what if the next arg starts with `-` like `-r`

Reply:
I prefer my original implementation, as this can cause fewer bugs IMO:
```js
const watchOptions = ['--watch', '--watch-path', '--watch-preserve-output'];

function argIsWatchOptionWithValue(arg) {
  return watchOptions.some((watchOption) => StringPrototypeStartsWith(arg, `${watchOption}=`));
}

function argIsWatchOption(arg) {
  return watchOptions.includes(arg) || argIsWatchOptionWithValue(arg);
}

const args = ArrayPrototypeFilter(process.execArgv, (arg, i, arr) => {
  // Don't pass --watch related args to children
  if (argIsWatchOption(arg)) {
    return false;
  }

  // First arg so it can't be a value of some watch option
  if (i === 0) {
    return true;
  }

  const prevArg = arr[i - 1];
  const prevArgIsWatchOption = argIsWatchOption(prevArg);

  // This is not a value of arg option as the previous arg is also not watch-related
  if (!prevArgIsWatchOption) {
    return true;
  }

  // If prev arg is watch related, but it has a value it means than this arg is not a value of such option
  return argIsWatchOptionWithValue(prevArg);
});
```

Reply:
the above code is over complicated IMHO. 
do you have an example of a bug my fix includes?

Reply:
nope, just a general feeling